distribution/lib/Standard/AWS/0.0.0-dev/src/S3.enso

```diff
@@ -0,0 +1,30 @@
+from Standard.Base import all
+
+import project.AWS_Credential.AWS_Credential
+import project.Errors.S3_Error
+import project.Internal.Auth
+
+polyglot java import software.amazon.awssdk.services.s3.S3Client
+polyglot java import software.amazon.awssdk.services.s3.model.S3Exception
+
+type S3
+    ## Gets the list of the S3 buckets.
+    @credentials AWS_Credential.default_widget
+    list_buckets : AWS_Credential | Nothing -> Vector Text ! S3_Error
+    list_buckets credentials:(AWS_Credential | Nothing)=Nothing = catch_s3 <|
+        client = make_client credentials
+        buckets = client.listBuckets.buckets
+        buckets.map b->b.name
+
+## PRIVATE
+catch_s3 : Function -> Any ! S3_Error
```

Comment:
```suggestion
handle_s3_errors : Function -> Any ! S3_Error
```

IMO the function name should be more descriptive. `catch_s3` sounds like it would be catching an "S3" "thing". Also it catches panics and converts them to dataflow so IMO `handle` will work, but I'm not as attached to it.